crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -44,6 +48,7 @@ pub struct TransactiondOrchestrator<A> {
     node_sync_handle: NodeSyncHandle,
     validator_state: Arc<AuthorityState>,
     _local_executor_handle: JoinHandle<()>,
+    pending_tx_log: Arc<FullNodePendingTransactionLog>,
```

Comment:
Why should the pending tx system be part of the orchestrator? To my understanding, the orchestrator is specifically for managing the connection between QD and fullnode, to enable local execution of QD transactions on the fullnode. However, the pending tx system is more tied to QD than orchestrator.
For example, if someone wants to use a lightweight QD on their own, they would want QD + pending tx management, but won't want the orchestrator.

Reply:
The more adaptive mindset if the orchestrator is a thing on top of quorum driver and does whatever things quorum driver itself does not do. Having this feature in quorum driver will turn it stateful, while having these things outside keeps the struct clean with what it does. 

I've thought about having this in QD (basically what you said `if someone wants to use a lightweight QD`). But i failed to convince myself this speculative benefits are worth it, like many other choices we made in the past. After all, we want clients to handle objects and retries by themselves, so if they want to use a lightweight QD they'd more than this component.

My plan with refactoring QD/Orchestrator is to have a queue in Orchestrator, so most of the heavy lifting would happen on Orchestrator. Eventually QD may just become a very lightweight wrapper for a few functions authority aggregator that clients to use as a lego block to build what they want

Reply:
What would you say is the difference between AuthorityAggregator and QuorumDriver?
I think AuthorityAggregator is the lightweight thing you are referring to, right?

Reply:
QD exposes only transaction submission logic (authority aggregator has a lot of things), and with reconfig-ability. It's the minimal work unit. One could argue authority aggregator is the minimal work unit, I wouldn't disagree if they don't mind extra work to stitch together some APIs and reconfig